File: src/WecErp.Infrastructure/ErpDbContextFactory.vb
Imports System
Imports Microsoft.EntityFrameworkCore
Imports Microsoft.EntityFrameworkCore.Design

Namespace WecErp.Infrastructure
    Public Class ErpDbContextFactory
        Implements IDesignTimeDbContextFactory(Of ErpDbContext)

        Public Function CreateDbContext(args As String()) As ErpDbContext Implements IDesignTimeDbContextFactory(Of ErpDbContext).CreateDbContext
            Dim connectionString = Environment.GetEnvironmentVariable("WECERP_DESIGN_CONNECTION")
            If String.IsNullOrWhiteSpace(connectionString) Then
                connectionString = "Server=(localdb)\MSSQLLocalDB;Database=WecErp_Dev;Trusted_Connection=True;TrustServerCertificate=True"
            End If

            Dim options = New DbContextOptionsBuilder(Of ErpDbContext)().
                UseSqlServer(connectionString).
                Options

            Return New ErpDbContext(options)
        End Function
    End Class
End Namespace

File: src/WecErp.Infrastructure.Migrations/ErpDbContextFactory.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Design;
using WecErp.Infrastructure;

namespace WecErp.Infrastructure.Migrations;

public sealed class ErpDbContextFactory : IDesignTimeDbContextFactory<ErpDbContext>
{
    public ErpDbContext CreateDbContext(string[] args)
    {
        var connectionString =
            Environment.GetEnvironmentVariable("WECERP_DESIGN_CONNECTION")
            ?? "Server=(localdb)\\MSSQLLocalDB;Database=WecErp_Dev;Trusted_Connection=True;TrustServerCertificate=True";

        var options = new DbContextOptionsBuilder<ErpDbContext>()
            .UseSqlServer(
                connectionString,
                sql => sql.MigrationsAssembly(typeof(ErpDbContextFactory).Assembly.FullName))
            .Options;

        return new ErpDbContext(options);
    }
}
